src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1729,3 +1730,79 @@ export const handleReorderFormField = [
   }),
   _handleReorderFormField,
 ] as RequestHandler[]
+
+/**
+ * NOTE: Exported for testing.
+ * Private handler for PUT /api/v3/admin/forms/:formId/collaborators
+ * @precondition Must be preceded by request validation
+ * @security session
+ *
+ * @returns 200 with updated collaborators and permissions
+ * @returns 403 when current user does not have permissions to update the collaborators
+ * @returns 404 when form cannot be found
+ * @returns 410 when updating collaborators for an archived form
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const _handleUpdateCollaborators: RequestHandler<
+  { formId: string },
+  PermissionsUpdateDto | ErrorDto,
+  PermissionsUpdateDto
+> = (req, res) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+  // Step 1: Get the form after permission checks
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve form with write permission check.
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Write,
+        }),
+      )
+      // Step 2: Update the form collaborators
+      .andThen((form) =>
+        AdminFormService.updateFormCollaborators(form, req.body.permissionList),
+      )
+      .map((updatedCollaborators) =>
+        res
+          .status(StatusCodes.OK)
+          .json({ permissionList: updatedCollaborators }),
```

Comment:
@karrui not in favour of fixing because idt it's a big issue but @mantariksh raised the concern that the shapes might be different - do you think it's important enough to fix?

Reply:
so this was the discussion yesterday; for my 2 cents I think anti makes sense that updating form collaborators return the new form collaborator array

should we change it to return an array of new collaborators?

Reply:
Thanks for the input! I'll update it accordingly